flocker/ca/functional/test_validation.py

```diff
@@ -74,6 +81,22 @@ def getContext(self):
         return ctx
 
 
+class WaitForDisconnectsFactory(ServerFactory):
+    """
+    A factory for use with ``SendingProtocol`` that makes it possible to wait
```

Comment:
We will have at least as much information as we have now, a dirty reactor warning pointing at the connection that's still connected.